core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -3839,7 +3848,20 @@ private void o3ProcessPartitionRemoveCandidates0(int n) {
                             false
                     );
                     TableUtils.txnPartitionConditionally(other, txn);
-                    long errno = ff.rmdir(other.$());
+                    other.$();
+                    if (ff.isSoftLink(other)) {
```

Comment:
You need to change `O3PartitionPurgeJob` to deal with soft links.
When partition cannot be deleted here because of opened `TableReader` it will schedule `O3PartitionPurgeJob` run to delete async and you need to treat there soft links in the same way.

Reply:
ah!, ok, indeed, having a look at it now